addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,379 @@
+            if home:
+                path = os.path.join(home(), 'odoo/addons/point_of_sale/tools/posbox/configuration/connect_to_server.sh')
+                try:
+                    subprocess.check_call([path, url, '', token, 'noreboot'])
+                    m.send_alldevices()
+                    image = os.path.join(home(), 'odoo/addons/hw_drivers/static/img/True.jpg')
+                except subprocess.CalledProcessError as e:
+                    _logger.warning(e.output)
+            else:
+                _logger.warning('Home environment variable doesn\'t exist')
+        if os.path.isfile(image):
+            with open(path, 'rb') as f:
+                return f.read()
 
 #----------------------------------------------------------
-# Bluetooth drivers
+# Drivers
 #----------------------------------------------------------
-btdrivers = []
 
-class BtMetaClass(type):
+drivers = []
+bt_devices = {}
+iot_devices = {}
+
+class DriverMetaClass(type):
     def __new__(cls, clsname, bases, attrs):
-        newclass = super(BtMetaClass, cls).__new__(cls, clsname, bases, attrs)
-        btdrivers.append(newclass)
+        newclass = super(DriverMetaClass, cls).__new__(cls, clsname, bases, attrs)
+        drivers.append(newclass)
         return newclass
 
+class Driver(Thread, metaclass=DriverMetaClass):
+    """
+    Hook to register the driver into the drivers list
+    """
+    connection_type = ""
 
-class BtDriver(Driver, metaclass=BtMetaClass):
-
-
-    def __init__(self, device, manager):
-        super(BtDriver, self).__init__()
+    def __init__(self, device):
+        super(Driver, self).__init__()
         self.dev = device
-        self.manager = manager
-        self.value = ''
+        self.data = {'value': ''}
         self.gatt_device = False
 
-    def disconnect(self):
-        path = "bt_%s" % (self.dev.mac_address,)
-        del drivers[path]
-
-    def get_name(self):
-        return self.dev.alias()
+    @property
+    def device_name(self):
+        return self._device_name
+
+    @property
+    def device_identifier(self):
+        return self._device_identifier
+
+    @property
+    def device_connection(self):
+        """
+        On specific driver override this method to give connection type of device
+        return string
+        possible value : direct - network - bluetooth
+        """
+        return self._device_connection
+
+    @property
+    def device_type(self):
+        """
+        On specific driver override this method to give type of device
+        return string
+        possible value : printer - camera - device
+        """
+        return self._device_type
+
+    @classmethod
+    def supported(cls, device):
+        """
+        On specific driver override this method to check if device is supported or not
+        return True or False
+        """
+        pass
 
-    def value(self):
-        return self.value
+    def get_message(self):
+        return ''
 
     def action(self, action):
```

Comment:
the signature changed
```suggestion
     def action(self, session_id, data):
         raise NotImplementedError()
```